### PR TITLE
fix(cli): Detect if running in cloudshell to output correct update command

### PR DESCRIPTION
**Pull request by dmurray-lacework**
issue: https://lacework.atlassian.net/browse/ALLY-567

When cli is running in cloudshell the update command should output:
`curl https://raw.githubusercontent.com/lacework/go-sdk/main/cli/install.sh | bash -s -- -d $HOME/bin
`

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: cli/cmd/cli_unix.go [hunk 1]
```diff
@@ -33,6 +33,13 @@ var promptIconsFunc = func(icons *survey.IconSet) {
 	icons.Question.Text = "▸"
 }
 
+// Env variables found in GCP, AWS and Azure cloudshell
+var (
+	gcpCloudEnv   = "CLOUD_SHELL"
+	awsCloudEnv   = "AWS_EXECUTION_ENV"
+	AzureCloudEnv = "POWERSHELL_DISTRIBUTION_CHANNEL"
```
Comment (rmoles) [inline cli/cmd/cli_unix.go:40]: Is this available both in the Azure powershell and bash terminals?

Comment (afiune) [reply cli/cmd/cli_unix.go:40]: tenor-44026482

Comment (dmurray-lacework) [reply cli/cmd/cli_unix.go:40]: It does indeed. Confirmed this yesterday. Salim raised the same question.

File: cli/cmd/cli_unix.go [hunk 2]
```diff
@@ -33,6 +33,13 @@ var promptIconsFunc = func(icons *survey.IconSet) {
 	icons.Question.Text = "▸"
 }
 
+// Env variables found in GCP, AWS and Azure cloudshell
+var (
```
Comment (afiune) [inline cli/cmd/cli_unix.go:37]: nit
```suggestion
const (
```

File: cli/cmd/cli_unix.go [hunk 3]
```diff
@@ -33,6 +33,13 @@ var promptIconsFunc = func(icons *survey.IconSet) {
 	icons.Question.Text = "▸"
 }
 
+// Env variables found in GCP, AWS and Azure cloudshell
```
Comment (afiune) [inline cli/cmd/cli_unix.go:36]: let us make sure that this comment explains the motivation/reasoning why we look for these variables

Comment (dmurray-lacework) [inline cli/cmd/cli_unix.go:36]:
```suggestion
// Env variables found in GCP, AWS and Azure cloudshell. Used to determine if cli is running on cloudshell.
```

File: cli/cmd/cli_unix.go [hunk 4]
```diff
@@ -41,7 +48,17 @@ func (c *cliState) UpdateCommand() string {
...
+  curl https://raw.githubusercontent.com/lacework/go-sdk/main/cli/install.sh | bash -s -- -d $HOME/bin
+`
+	}
 	return `
   curl https://raw.githubusercontent.com/lacework/go-sdk/main/cli/install.sh | bash
 `
 }
+
+func isCloudshell() bool {
+	return os.Getenv(gcpCloudEnv) == "true" || os.Getenv(awsCloudEnv) == "Cloudshell" || os.Getenv(AzureCloudEnv) == "CloudShell"
```
Comment (afiune) [inline cli/cmd/cli_unix.go:63]: Could we split these checks into three different functions?
* `isAwsCloudShell()`
* `isGcpCloudShell()`
* `isAzureCloudShell()`

These functions has to explain what it does (check if running on XYZ cloud shell) and what to we expect.

Then this function should call them all:
```suggestion
	return isAwsCloudShell() || isGcpCloudShell() || isAzureCloudShell()
```

**Review by afiune — COMMENTED** [review]
write a few tests to verify this behavior please 🙌🏽

**Review by afiune — APPROVED** [review]
tenor-179775173

**Pull request merged**